docs/new_api.md

```diff
@@ -0,0 +1,646 @@
+# New API Detailed Description
+This design is based on moving towards an API that is more RESTful, and attempting to keep request/response patterns consistent and predictable.
+
+## Contents
+1. [Standard Query Parameters](#standard-query-parameters)
+2. [Standard Request Body](#standard-request-body)
+3. [Standard Response Body](#standard-response-body)
+4. [Error Response](#error-response)
+5. [Accept and Content-Type Headers](#accept-and-content-type-headers)
+6. [Authentication and Authorization](#authentication-and-authorization)
+7. [Activities](#activities)
+8. [CBs](#cbs)
+9. [Visitors](#visitors)
+
+## Standard Query Parameters
+In general, filters should be supported through query params in `GET` requests (where such functionality is appropriate) by allowing the client to specify key-value pairs as query parameters, where the key corresponds to a field in the response body.
+
+For example, in `GET /api/visitors`, the response body is:
+```
+{ result: [{ id, name, gender, email, ... }] }
+```
+We should specify a filter on `gender` and `email` in the following way:
+```
+GET /visitors?gender=foo&email=bar
+```
+
+In addition, the following query parameters should be standard:
```

Comment:
When you stay they're standard does this mean that every get request should have all of these params?

Reply:
It means that you should be able to pass these parameters to every endpoint and for them to have the desired effect if the parameter is meaningful for that endpoint.

Reply:
Surely it's not necessary to write functionality for this is on all `GET` requests. For example on the feedback request will never need to be sorted as there is no tabulated data.

Reply:
We write a function to do this once, and can use it everywhere.